mux/src/pane.rs

```diff
@@ -192,6 +373,37 @@ pub trait Pane: Downcast + Send + Sync {
 
     fn get_logical_lines(&self, lines: Range<StableRowIndex>) -> Vec<LogicalLine>;
 
+    fn get_logical_lines_raw(&self, lines: Range<StableRowIndex>) -> String {
```

Comment:
I don't believe that we need this function or the complicated `diff_attrs` function above at all. Look at:

https://github.com/wez/wezterm/blob/22f9f8d288521508bd4cdc924d9f209d78b36483/lua-api-crates/termwiz-funcs/src/lib.rs#L268

which is how we render a set of `Line` into ansi escape sequences to support that various lua functions like https://wezfurlong.org/wezterm/config/lua/pane/get_lines_as_escapes.html

Reply:
Ah okay, that'll make life easier. When I started on this the lua function didn't exist yet.